### PR TITLE
Refactor 01-backup

### PR DESCRIPTION
**Pull request by douglatornell**
Split the monolithic 01-backup lesson section into 8 smaller topic sections.

I did a lot of this refactoring for a workshop that I created in Jun-2015 based on the SWC material. it is rendered at http://douglatornell.ca/training/meopeers-2015-06-15/

We are going to teach this refactored version of the hg lesson next week at the UBC EOAS workshop, so it will get trial by fire, but I would also appreciate reviewer comments.

### CODE REVIEW

File: instructors.html [hunk 1]
```diff
@@ -28,15 +28,22 @@
...
-<p>The solution is to use <a href="reference.html#version-control">version control</a> to manage their work. Version control is better than mailing files back and forth because:</p>
+          <p>Using a software tool to handle the versions of your project files lets you focus on the more interesting/innovative aspects of your project.</p>
+<ul>
+<li>Version control’s advantages:
 <ul>
 <li>Nothing that is committed to version control is ever lost. This means it can be used like the “undo” feature in an editor, and since all old versions of files are saved it’s always possible to go back in time to see exactly who wrote what on a particular day, or what version of a program was used to generate a particular set of results.</li>
```
Comment (jdcorless) [inline instructors.html:35]: I think the clause "it’s always possible to go back in time to see exactly who wrote what on a particular day" is redundant with the next list item

File: 02-configuration.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+>
+> * Explain what configuration steps are required the first time Mercurial
+>   is used on a computer.
+
+When we use Mercurial on a computer for the first time,
+we need to configure a few things.
+The command
+
+~~~ {.bash}
+$ hg config --edit
```
Comment (jdcorless) [inline 02-configuration.md:17]: On a new Ubuntu VM with hg ver 2.8.2 installed, this command did not work (hg: unknown command 'config').  I think in the old hg lesson we just had students editing the config file manually.

File: 02-configuration.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+# (see "hg help extensions" for more info)
+#
+# pager =
+# progress =
+# color =
+~~~
+
+Edit the file to set `username` to your own name and email address and add your
+favourite editor in the `[ui]` section.
+Also uncomment the the `pager =`,
```
Comment (jdcorless) [inline 02-configuration.md]: typo

File: 02-configuration.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+[color]
+mode = win32
+~~~
+
+When you are finished,
+save the file and exit your editor.
+You can check that your settings are the way that you want them with the command
+
+~~~ {.bash}
+$ hg config ui extensions
```
Comment (jdcorless) [inline 02-configuration.md:124]: Same issue (hg: unknown command 'config')

File: 02-configuration.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+ui.username=Doug Latornell <djl@douglatornell.ca>
+~~~
+
+When you saved the file stored in your home directory as `.hgrc` on OS X and
+Linux or as `Mercurial.ini` on Windows.
+The fact that these settings are in the Mercurial configuration file in our home
+directory means that they will be used for every Mercurial repository on this
+machine.
+
+The above configuration work only needs to be done tghe first time you use
```
Comment (jdcorless) [inline 02-configuration.md]: typo

File: 04-tracking.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+That permanent copy is called a [commit](reference.html#commit)
+(or [revision](reference.html#revision)).
+
+We use the `-m` flag (for "message") to record a comment that will help us remember later on what we did and why.
+If we just run `hg commit` without the `-m` option,
+Mercurial will launch `nano`
+(or whatever other editor we configured at the start)
+so that we can write a longer message.
+
+[Good commit messages][commit-messages] start with a brief (<70 characters) summary of
```
Comment (jdcorless) [inline 04-tracking.md]: Cool idea to link to this as reference material for students!

File: 02-configuration.md (hunk 2 above)
Comment (douglatornell) [inline 02-configuration.md:17]: Good catch! Looks like `hg config --edit` was introduced in Mercurial v3.0 (2014-05-01).

Yes, the previous version had direct editing of the config file. `hg config --edit` avoids the diversion of where the config file is on different platforms and what it is called. Would it work to put the direct editing approach in a callout to be used in the cases where `hg config --edit` fails?

File: 04-tracking.md (hunk 6 above)
Comment (douglatornell) [inline 04-tracking.md]: Credit to the git-novice lesson for reminding me of that link in this round of edits, though I know I've liked to it in other material.

On the topic of reference links, I like to include a richer collection on the index page than the generic SWC lesson does (Reference, Discussion, Instructor's Guide). I usually add the Mercurial site, the Mercurial - The Definitive Guide book site, and a collection of "[command reminders](https://douglatornell.github.io/hg-novice-2015-09-22-ubc/commands.html)". Your comment makes me think that the Good Commit Messages link would be a good addition to that.

What do you think about making that collection of links a standard thing on the index page for the Mercurial lesson?

File: 02-configuration.md (hunk 2 above)
Comment (jdcorless) [inline 02-configuration.md:17]: I think that makes sense, especially if your install instructions get most users set up with hg v3.0 and then the callout is just for those who decided not to upgrade.

File: 04-tracking.md (hunk 6 above)
Comment (jdcorless) [inline 04-tracking.md]: I really like that idea.  I think that we want to leave students with essentially a curated list of where to go to build on the intro information they've just gone through.  Web searching is great but nothing like an expert encouraging you in the right direction!

File: 01-basics.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+We'll start by exploring how version control can be used
+to keep track of what one person did and when.
+Even if you aren't collaborating with other people,
+automated version control is much better than this situation:
+
+[![Piled Higher and Deeper by Jorge Cham, http://www.phdcomics.com](fig/phd101212s.gif)](http://www.phdcomics.com)
+
+"Piled Higher and Deeper" by Jorge Cham, http://www.phdcomics.com
+
+We've all been in this situation before: it seems ridiculous to have multiple nearly-identical versions of the same document. Some word processors let us deal with this a little better, such as Microsoft Word's "Track Changes" or Google Docs' version history.
```
Comment (kynan) [inline 01-basics.md]: Minor niggle: would be good to be consistent about line length: either wrap or don't

File: 01-basics.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+Version control systems start with a base version of the document and then save just the changes you made at each step of the way. You can think of it as a tape: if you rewind the tape and start at the base document, then you can play back each change and end up with your latest version.
+
+![Changes are saved sequentially](fig/play-changes.svg)
+
+Once you think of changes as separate from the document itself, you can then think about "playing back" different sets of changes onto the base document and getting different versions of the document. For example, two users can make independent sets of changes based on the same document.
+
+![Different versions can be saved](fig/versions.svg)
+
+If there aren't conflicts, you can even try to play two sets of changes onto the same base document.
```
Comment (kynan) [inline 01-basics.md]: are no conflicts

File: 04-tracking.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+> * Display the history of changes to files in a repository and explain the metadata that is recorded with each changeset.
+
+Let's create a file called `plan.txt` in which Susan is going to write her
+initial ideas and notes about the Salish Sea NEMO daily forecast system.
+You can use any text editor you want
+(`Notepad`, `TextEdit`, `gedit`, `nano`, `emacs`, `vi`, ...)
+but please don't use a word processor like Microsoft Word or LibreOffice Write.
+In particular,
+the editor does not have to be the one that you set in your Mercurial
+configuration earlier.)
```
Comment (kynan) [inline 04-tracking.md]: Extra )

File: 07-revert.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+> ~~~
+>
+> to our Mercurial configuration settings.
+> After we save the configuration file and exit from the editor we have a new
+> command,
+> `hg nb-revert`
+> (no backup revert) that is the same as typing `hg revert --no-backup`.
+> Note that it is good practice when creating aliases for commands to give them
+> your own names instead of using the alias to redefine the built-in command.
+If we had used the alias to change add the `--no-backup` flag to the `revert`
```
Comment (kynan) [inline 07-revert.md]: `>` missing

File: index.md [hunk 11]
```diff
@@ -17,17 +17,30 @@ can and should be stored in a version control system.
...
-    so that if people have questions later on
-    they know who to ask.
-*   It's hard (but not impossible) to accidentally overlook or overwrite someone's changes:
-    the version control system automatically notifies users
-    whenever there's a conflict between one person's work and another's.
+
+*   As we have this record of who made what changes when,
+    we know who to ask if we have questions later on,
+    and,
+    if needed it,
```
Comment (kynan) [inline index.md]: remove "it"

**Comment by kynan** [comment]
Apart from minor niggles this is looking good I think.

2 general remarks:
- It would be good to be consistent about line length for paragraphs: either break at a fix margin (e.g. 80 columns) or not at all.
- Do we want to use the informal style (let's, don't) as it is now?

File: instructors.html (hunk 1 above)
Comment (douglatornell) [inline instructors.html:35]: I'll include this in the rewrite of 1`instructors.md`.

**Comment by kynan** [comment]
@douglatornell: seems this has some conflicts now.

**Comment by douglatornell** [comment]
Yeah, I noticed that when I was working on it last week. Will try to
resolve them this week.

Does anyone have any advice/opinions on whether I should merge this into
gh-pages, or into a new working branch for the v5.4 lesson release? I have
more material coming for the v5.4 release, and would like to have the
complete new lesson available somewhere before the 13-Jun Bug BBQ. But I'm
not sure if the material should go into the gh-pages branch before then.

**Comment by gvwilson** [comment]
I suggest putting it into gh-pages.

**Comment by douglatornell** [comment]
Okay. No worries about messing up instructor/workshops between now and the
v5.4 release?

**Comment by gvwilson** [comment]
I don't think we have any Mercurial workshops scheduled between now and 
then.

File: 02-configuration.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+directory means that they will be used for every Mercurial repository on this
+machine.
+
+The above configuration work only needs to be done the first time you use
+Mercurial on a computer.
+
+You can check that your settings are the way that you want them with the command
+
+~~~ {.bash}
+$ hg config ui extensions
```
Comment (kynan) [inline 02-configuration.md:124]: This will again fail in hg pre 3.0

File: 01-basics.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+> ## The long history of version control systems {.callout}
+>
+> Automated version control systems are nothing new. Tools like RCS, CVS, or
+> Subversion are considered now legacy systems, offering more limited
+> capabilities than modern tools, such as
+> [Mercurial](http://mercurial.selenic.com/) and
+> [Git](http://swcarpentry.github.io/git-novice/).
+> In particular,
+> the latter are *distributed*, meaning that they don't need a centralized
+> server to host the repository.
```
Comment (kynan) [inline 01-basics.md:65]: Instead, every user has their own full copy of the repository.

**Pull request merged**